src/generic/stage2/numberparsing.h

```diff
@@ -179,26 +167,12 @@ simdjson_really_inline bool compute_float_64(int64_t power, uint64_t i, bool neg
   // which we guard against.
   // If we have lots of trailing zeros, we may fall right between two
   // floating-point values.
-  if (simdjson_unlikely((lower == 0) && ((upper & 0x1FF) == 0) &&
+  // We have 5**27 < 2**64 and it is the largest power of 5 to do so.
+  if (simdjson_unlikely((lower == 0) && (power >= -27) && (power <= 27)  &&
```

Comment:
The inference here is that any number >= 2^64 cannot possibly be between two values because it's got too many significant digits to do so. I'm curious why that's not 2^54?

Also, double space before the && -> single space, unless this was on purpose?

Reply:
> The inference here is that any number >= 2^64 cannot possibly be between two values because it's got too many significant digits to do so. I'm curious why that's not 2^54?

This part has been explained much better now...

```
  // 
  // The round-to-even cases take the form of a number 2m+1 which is in (2^53,2^54]
  // times a power of two. That is, it is right between a number with binary significand
  // m and another number with binary significand m+1; and it must be the case
  // that it cannot be represented by a float itself.
  //
  // We must have that w * 10 ^q == (2m+1) * 2^p for some power of two 2^p.
  // Recall that 10^q = 5^q * 2^q.
  // When q >= 0, we must have that (2m+1) is divible by 5^q, so 5^q <= 2^54. We have that
  //  5^23 <=  2^54 and it is the last power of five to qualify, so q <= 23.
  // When q<0, we have  w  >=  (2m+1) x 5^{-q}.  We must have that w<2^{64} so
  // (2m+1) x 5^{-q} < 2^{64}. We have that 2m+1>2^{53}. Hence, we must have 
  // 2^{53} x 5^{-q} < 2^{64}.
  // Hence we have 5^{-q} < 2^{11}$ or q>= -4$  (64-bit case). 
  //
  // We require lower <= 1 and not lower == 0 because we could not prove that 
  // that lower == 0 is implied; but we could prove that lower <= 1 is a sufficient test.
```

I still need to document this better, but there is only so much I can put in comments.

Reply:
I have removed the double space.